src/sentry/models/event.py

```diff
@@ -633,6 +619,17 @@ class Meta:
 
     __repr__ = sane_repr("project_id", "group_id")
 
+    def __init__(self, *args, **kwargs):
+        node_data = kwargs.pop("node_data", None)
```

Comment:
I think having both node_data and data is a bit confusing, thus it is hard to assess what the behavior is.
What's the difference between data and node_data ? When would we use one or the other ?

Reply:
Event accepts either data or NodeData now, I updated this to check the type instead

Reply:
I think my questions is more about the which use cases should pass data and which ones should pass node_data.

Reply:
Right now only event manager passes node data. It's kind of a hack to avoid further refetching node data during post process, since it gets a NodeData with no node_id it won't fetch from nodestore anymore.

Reply:
Why do we need to avoid refetching node data in post process?